src/OpenTelemetry.Api/Context/Propagation/TraceContextPropagator.cs

```diff
@@ -230,18 +237,72 @@ internal static bool TryExtractTracestate(string[] tracestateCollection, out str
 
             if (tracestateCollection != null)
             {
-                var result = new StringBuilder();
+                List<string> listMembers = new List<string>();
+                for (int i = 0; i < tracestateCollection.Length; ++i)
+                {
+                    // test_tracestate_multiple_headers_different_keys
+                    listMembers.AddRange(tracestateCollection[i].Split(','));
+                }
+
+                if (listMembers.Count > 32)
+                {
+                    // https://github.com/w3c/trace-context/blob/master/spec/20-http_request_header_format.md#list
+                    // test_tracestate_member_count_limit
+                    return false;
+                }
 
-                // Iterate in reverse order because when call builder set the elements is added in the
-                // front of the list.
-                for (int i = tracestateCollection.Length - 1; i >= 0; i--)
+                var keySet = new HashSet<string>();
+                var result = new StringBuilder();
+                for (int i = 0; i < listMembers.Count; ++i)
                 {
-                    if (string.IsNullOrEmpty(tracestateCollection[i]))
+                    // https://github.com/w3c/trace-context/blob/master/spec/20-http_request_header_format.md#tracestate-header-field-values
+                    if (string.IsNullOrEmpty(listMembers[i]))
+                    {
+                        // Empty and whitespace - only list members are allowed.
+                        // Vendors MUST accept empty tracestate headers but SHOULD avoid sending them.
+                        continue;
+                    }
+
+                    // Spaces and horizontal tabs surrounding list-members are ignored.
+                    var listMember = listMembers[i].Trim(new char[] { ' ', '\t' });
```

Comment:
Let's promote `new char[] { ' ', '\t' }` to a static so we can reuse it.

Reply:
Done.